pallets/parachain-staking/src/tests.rs

```diff
@@ -4528,6 +4500,307 @@ fn delegation_events_convey_correct_position() {
 		});
 }
 
+#[test]
+fn no_rewards_paid_until_after_reward_payment_delay() {
+	ExtBuilder::default()
+		.with_balances(vec![(1, 20), (2, 20), (3, 20), (4, 20)])
+		.with_candidates(vec![(1, 20), (2, 20), (3, 20), (4, 20)])
+		.build()
+		.execute_with(|| {
+			roll_to_round_begin(2);
+			// payouts for round 1 
+			set_author(1, 1, 1);
+			set_author(1, 2, 1);
+			set_author(1, 3, 1);
+			set_author(1, 4, 1);
+			set_author(1, 4, 1);
+			let mut expected = vec![
+				Event::CollatorChosen(2, 1, 20),
+				Event::CollatorChosen(2, 2, 20),
+				Event::CollatorChosen(2, 3, 20),
+				Event::CollatorChosen(2, 4, 20),
+				Event::NewRound(5, 2, 4, 80),
+			];
+			assert_eq_events!(expected);
+
+			roll_to_round_begin(3);
+			expected.append(&mut vec![
+				Event::CollatorChosen(3, 1, 20),
+				Event::CollatorChosen(3, 2, 20),
+				Event::CollatorChosen(3, 3, 20),
+				Event::CollatorChosen(3, 4, 20),
+				Event::NewRound(10, 3, 4, 80),
+				// rewards will begin immediately following a NewRound
+				Event::Rewarded(3, 1),
```

Comment:
Collator 3 gets paid first because payments are in order of storage hash two_x(account_id)